src/components/newContractor/NewContractorForm.js

```diff
@@ -0,0 +1,134 @@
+import React, { useContext} from 'react';
+import PropTypes from 'prop-types'
+import { useFormik } from 'formik'
+import * as Yup from 'yup'
+
+import Card from '../layout/card/Card'
+import Input from '../layout/form/Input'
+import Button from '../layout/button/Button'
+import AlertContext from '../../context/alert/alertContext'
+
+import './newContractorForm.css'
+
+const NewContractorForm = (props) => {
+
+    // variables for Alert display - AlertContext
+    const alertCtx = useContext(AlertContext)
+    const { setAlertMessage } = alertCtx
+
+        const formik = useFormik({
+        initialValues: {
+            companyName: '',
+            NIP: '',
+            street: '',
+            post: '',
+            city: '',
+            tel: ''
+        },
+
+        validationSchema: Yup.object({
+            companyName: Yup.string()
+                .max(30, "Maksymalnie 30 znaków")
+                .required('Required'),
+            NIP: Yup.string()
+                .required('Required'),
+            street: Yup.string()
+                .max(30, "Maksymalnie 30 znaków")
+                .required('Required'),
+            post: Yup.string()
+                .max(10, "Maksymalnie 10 znaków")
+                .required('Required'),
+            city: Yup.string()
+                .max(30, "Maksymalnie 30 znaków")
+                .required('Required'),
+            tel: Yup.string()
+                .max(14, "Maksymalnie 14 znaków")
+                .required('Required'),
+        }),
+
+        onSubmit: values => {
+            console.log(values)
+            fetch(`http://localhost:5000/contractors/`,
+            {method: 'POST',
+            headers: { "Content-type": "application/json" },
+            body: JSON.stringify({...values})
+            })
+            .then(res=> res.json())
+            .then(res => {
+                setAlertMessage('Kontrahent został dodany', 'pass')
+                props.setUpdateedContractorList()
+                props.closeForm()
+                formik.resetForm()
```

Comment:
I guess you don't have to `resetForm()` after you `closeForm()`. This may also trigger an error about updating state of unmounted component.